### PR TITLE
feat: add data mapping to a code preview

### PR DESCRIPTION
**Pull request by eljefe223**
# Description
Move code mapping from fast dev site to fast tooling.
Closes: #1779 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-tooling-react/src/data-utilities/mapping.ts [hunk 1]
```diff
@@ -24,6 +24,113 @@ import {
 /**
  * This file contains all functionality for mapping data
  */
+export interface CodePreview {
+    componentName: string;
+    childNames: CodePreviewChildOption[];
```
Comment (janechu) [inline packages/fast-tooling-react/src/data-utilities/mapping.ts]: can you rename `childNames` to `childOptions`

File: packages/fast-tooling-react/src/data-utilities/mapping.ts [hunk 2]
```diff
@@ -25,6 +25,113 @@ import {
...
+                childOptions
+            );
+        }, mappedData);
+}
+
+/**
+ * Maps data returned from the form generator to the code Preview
+ */
+export function mapDataToCodePreview(codePreview: CodePreview): string {
+    let cp: string = "TBD";
```
Comment (nicholasrice) [inline packages/fast-tooling-react/src/data-utilities/mapping.ts]: Why do we assign this "TBD"?

File: packages/fast-tooling-react/src/data-utilities/mapping.ts [hunk 3]
```diff
@@ -25,6 +25,113 @@ import {
...
+    variables = "";
```
Comment (nicholasrice) [inline packages/fast-tooling-react/src/data-utilities/mapping.ts]: Why are we assigning this an empty string and then concatenating it with the result of `generateReactCodePreview`? Doesn't that do nothing?

File: packages/fast-tooling-react/src/data-utilities/mapping.ts [hunk 4]
```diff
@@ -25,6 +25,113 @@ import {
...
+export interface ReactComponentConfig {
+    hasChildren: boolean;
+    componentName: string;
+    componentData: any;
+    tab: string;
+    newTab: string;
+    location: string;
+}
+
+let variables: string;
```
Comment (nicholasrice) [inline packages/fast-tooling-react/src/data-utilities/mapping.ts]: What is this variable for?

**Pull request merged**